baby-gru/src/components/validation-tools/MoorhenCarbohydrateValidation.tsx

```diff
@@ -0,0 +1,182 @@
+                />
+                </div>
+            )
+        })}
+
+        <p style={{margin: "auto"}}><i>Powered by Privateer</i></p>
+    </>
+
+}
+
+export const MoorhenCarbohydrateValidation = (props: Props) => {
+
+    const isDark = useSelector((state: moorhen.State) => state.sceneSettings.isDark)
+    const mapSelectRef = useRef<undefined | HTMLSelectElement>();
+    const moleculeSelectRef = useRef<undefined | HTMLSelectElement>();
+
+    const molecules = useSelector((state: moorhen.State) => state.molecules)
+    const maps = useSelector((state: moorhen.State) => state.maps)
+    const [selectedModel, setSelectedModel] = useState<null | number>(null)
+    const [selectedMap, setSelectedMap] = useState<null | number>(null)
+
+    const [privateerData, setPrivateerData] = useState<null | PrivateerResultsEntry[]>(null)
+
+
+    const handleModelChange = (evt: React.ChangeEvent<HTMLSelectElement>) => {
+        setSelectedModel(parseInt(evt.target.value))
+    }
+
+    const handleMapChange = (evt: React.ChangeEvent<HTMLSelectElement>) => {
+        setSelectedMap(parseInt(evt.target.value))
+    }
+
+    const dispatch = useDispatch()
+    const handleHover = useCallback((residueLabel: string) => {
+        if (selectedModel !== null) {
+            const molecule = molecules.find(item => item.molNo === selectedModel)
+            if (molecule) {
+                const [chain, resName, resNum] = residueLabel.split('/')
+                const cid = `//${chain}/${resNum}(${resName})`
+                dispatch(setHoveredAtom({molecule, cid}))
+            }
+        }
+    }, [selectedModel, molecules])
+
+
+    const handleClick = useCallback((residueLabel: string) => {
+        if (selectedModel !== null) {
+            const molecule = molecules.find(item => item.molNo === selectedModel)
+            if (molecule) {
+                const [chain, resName, resNum] = residueLabel.split('/')
+                const cid = `//${chain}/${resNum}(${resName})`
+                molecule.centreOn(cid)
+            }
+        }
+    }, [selectedModel, molecules])
+
+    useEffect(() => {
+        if (molecules.length === 0) {
+            setSelectedModel(null)
+        } else if (selectedModel === null) {
+            setSelectedModel(molecules[0].molNo)
+        } else if (!molecules.map(molecule => molecule.molNo).includes(selectedModel)) {
+            setSelectedModel(molecules[0].molNo)
+        }
+
+    }, [molecules.length])
+
+
+    const validate = async () => { 
+        const selectedMolecule = molecules.find(molecule => molecule.molNo === selectedModel)
+
+        if (selectedMolecule) {
+
+            const atoms = await selectedMolecule.getAtoms()
+            const privateerResult = await props.commandCentre.current.cootCommand({
+                command: 'shim_privateer_validate',
+                commandArgs: [atoms, selectedMolecule.name],
+                returnType: 'privateer_results'
+            }, false)
```

Comment:
We could change `shim_privateer_validate` so that a shim function is not needed and so that all it takes is the `selectedMolecule.molNo` as input. More on this in a comment I've made in `privateer-wrapper.cc`